From b4b419d3bb0d66b8878173461b06b5b4d8d6ba75 Mon Sep 17 00:00:00 2001
From: Gabriele M <moto.falcon.git@gmail.com>
Date: Sun, 15 Jul 2018 17:15:47 +0200
Subject: [PATCH 17/20] extract_utils: Skip unneeded md5sum

This was once used as we kept the extracted files after the end of
the extraction. We now use it to prevent multiple extractions within
the same session, but all we need is checking the existance of the
output directory.

Change-Id: I4192b2a69e5821300df879093038b7a27465b246
---
 build/tools/extract_utils.sh | 12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)

diff --git a/build/tools/extract_utils.sh b/build/tools/extract_utils.sh
index 29c62bdc..ef5cfd9a 100644
--- a/build/tools/extract_utils.sh
+++ b/build/tools/extract_utils.sh
@@ -936,18 +936,12 @@ function extract() {
     fi
 
     if [ -f "$SRC" ] && [ "${SRC##*.}" == "zip" ]; then
+        # We might call extract() multiple times (e.g., commonized trees).
+        # If $DUMPDIR exists, then we already extracted the image.
         DUMPDIR="$TMPDIR"/system_dump
-
-        # Check if we're working with the same zip that was passed last time.
-        # If so, let's just use what's already extracted.
-        MD5=`md5sum "$SRC"| awk '{print $1}'`
-        OLDMD5=`cat "$DUMPDIR"/zipmd5.txt`
-
-        if [ "$MD5" != "$OLDMD5" ]; then
-            rm -rf "$DUMPDIR"
+        if [ ! -d "$DUMPDIR" ]; then
             mkdir "$DUMPDIR"
             unzip "$SRC" -d "$DUMPDIR"
-            echo "$MD5" > "$DUMPDIR"/zipmd5.txt
 
             # Stop if an A/B OTA zip is detected. We cannot extract these.
             if [ -a "$DUMPDIR"/payload.bin ]; then
-- 
2.17.1

